app2.js

```diff
@@ -0,0 +1,79 @@
+// Get the form and tbody elements 
+const form = document.querySelector('#form')
+const tableBody = document.querySelector('#table-body')
+
+// Add a new book in the table
+function addBookToUI(book){
+  const tr = document.createElement('tr')
+  tr.innerHTML=
+  `
+  <td>${book.title}</td>
+  <td>${book.author}</td>
+  <td>${book.isbn}</td>
+  <td id="delete"><i class="fa-solid fa-trash text-danger delete" style="cursor:pointer"></i></td>
+  `
+  tableBody.appendChild(tr)
+
+  // Delete a book from the table
+  const delBtn = tr.querySelector('#delete') 
+  deleteBook(delBtn)
+}
+
+// Function to clear the input fields
+function clearFields(){
+    form.reset()
+}
+
+// Submit the form data
+form.addEventListener('submit', (e)=>{
+    e.preventDefault()
+    const title  = document.querySelector('#title').value 
+    const author = document.querySelector('#author').value
+    const isbn = document.querySelector('#ISBN').value 
+
+    const book = {
+        title: title.trim(),
+        author: author.trim(),
+        isbn: isbn.trim(),
+    }
+    if (!book.title || !book.author || !book.isbn){
+        alerts('blank')
+    }else{
+        addBookToUI(book)
+        alerts('success')
+        clearFields()
+    }
+    
+}) 
+// Function to perform the delete action 
+function deleteBook(target){
+    target.addEventListener('click', (e)=>{
```

Comment:
Implement the delete function such that it is not dependent to the addToUI function. The deleteBook needs to be called on an html element like the tbody